src/util/parsable_string.h

```diff
@@ -0,0 +1,100 @@
+// Copyright 2019 Diffblue Limited.
+
+/// \file
+/// A wrapper for strings that facilitates parsing
+
+#ifndef CPROVER_UTIL_PARSABLE_STRING_H
+#define CPROVER_UTIL_PARSABLE_STRING_H
+
+#include <stdexcept>
+#include <string>
+
+class parsable_stringt
+{
+private:
+  const std::string &underlying;
+  std::size_t start_pos;
+  std::size_t after_end_pos;
+
```

Comment:
Note that this mostly re-invents `std::string_view`.
I would suggest that we pull in an implementation of that, and use it.

Reply:
First Google result is this one: https://github.com/martinmoene/string-view-lite/blob/master/include/nonstd/string_view.hpp
Seem OK to pull that in?
With others we've renamed to string_viewt, but that will make forward-porting harder when we upgrade - should we leave the name alone?
Where are files like this (Boost licence) kept?

Reply:
I would use an alias, like we do with `optionalt`. I was going to suggest putting the file in the same place as `nonstd::optionalt` but the licence is different, so I'm not sure. Maybe @smowton or @tautschnig knows?

Reply:
That's funny, `nonstd::optional-lite` is also Boost licence. Hmm. It seems it used to be MIT-licenced, and it was re-licenced in June 2018: https://github.com/martinmoene/optional-lite/commit/03f75a98c48eb9e88d04e0995c69eb1baf5e9f2f. It seems that `nonstd::string_view-lite` was also re-licenced at the same time: https://github.com/martinmoene/string-view-lite/commit/3aed8744ee6df0266e79c4a4b568ba9800f2fbf8. If it's easier, perhaps it would be legal to copy it the version just before it was re-licenced?